src/TypeShape.TestCases/README.md

```diff
@@ -0,0 +1,20 @@
+# TypeShape consumer test cases
+
+This package is intended for consumption by your unit test project. It presents a bunch of types that help to exhaustively test your TypeShape visitors.
+
+```cs
+private static readonly IProviderUnderTest Provider = SourceGenProviderUnderTest.Default; // or RefectionProviderUnderTest.Default;
+
+[Theory]
+[MemberData(nameof(TestTypes.GetTestCases), MemberType = typeof(TestTypes))]
+public void Roundtrip_Value<T>(TestCase<T> testCase)
```

Comment:
Is it possible to use the `TestCase<T, TProvider>` as the theory parameter? I suspect it won't work.

Reply:
It apparently _does_.

Reply:
Oh interesting. Nice.

Reply:
This compiles and runs tests:

```cs
[Theory]
[MemberData(nameof(TestTypes.GetTestCases), MemberType = typeof(TestTypes))]
public void Roundtrip_Value<T, TProvider>(TestCase<T, TProvider> testCase)
	where TProvider : IShapeable<T>
{
	ITypeShape<T> shape = TProvider.GetShape();
```

Reply:
The only change this would make to the package I'm building here is to its README file. I could change it to prescribe these as the recommended pattern, but as the user can already get an `ITypeShape<T>` with the prior/current pattern, is there a benefit that I'm missing from having the user add `TProvider` as a type parameter?